internal/providers/aws/aws.go

```diff
@@ -72,8 +73,21 @@ func NewFetcher(l *log.Logger) (resource.Fetcher, error) {
 
 // Init prepares the fetcher for this platform
 func Init(f *resource.Fetcher) error {
-	// Determine the partition and region this instance is in
-	regionHint, err := ec2metadata.New(f.AWSSession).Region()
+	var regionHint string
+	// During the fetch stage we might be running before the networking is
+	// fully ready. Add a dumb retry (up to 30 seconds worth of retrying).
+	// 30 seconds was chosen entirely arbitrarily, normal HTTP calls would
+	// use a similar method with an infinite timeout but that shouldn't
+	// be done here unless the error detection is updated.
+	//
+	// TODO: investigate if specifying a Retryer & MaxRetries in the aws.Config
+	// would be sufficient to remove the need for a retry function.
+	err := util.RetryWithBackoff(100*time.Millisecond, 5*time.Second, 30*time.Second, func() error {
+		var innerErr error
+		// Determine the partition and region this instance is in
+		regionHint, innerErr = ec2metadata.New(f.AWSSession).Region()
```

Comment:
This doesn't respect the `Offline` flag.  Won't it retry uselessly during `fetch-offline`?

Reply:
Ok, in the latest patch set I've switched up from a dumb retry loop of `ec2metadata.New(f.AWSSession).Region()` to a `FetchToBuffer` call against the `/latest` metadata URL. By proxy of that it will now respect the `Offline` flag (also left a comment that if we change that in the future we need to consider the flag).